include/envoy/http/alternate_protocols_cache.h

```diff
@@ -109,5 +113,38 @@ class AlternateProtocolsCache {
   virtual size_t size() const PURE;
 };
 
+using AlternateProtocolsCacheSharedPtr = std::shared_ptr<AlternateProtocolsCache>;
+
+/**
+ * A manager for multiple alternate protocols caches.
+ */
+class AlternateProtocolsCacheManager {
```

Comment:
two thoughts on docs.

optionally, as you and I flailed a bit sorting out how the dns cache worked, it'd be cool (but optional) to start documenting how this works for developers and tossing it in source/docs quic_upstream.md or some such.  

Also optional (but this one you have to do eventually) we're going to need to beef up the user docs on how upstream http/3 works, and if we stash that temporarily in source/docs it'd be a helpful road map for non-alyssa secondary reviewers as we land PRs in this space.

Reply:
I'm not the best with docs, but I've taken a stab in http3_upstream.md. Is this what you were thinking? (I named it HTTP/3 instead of QUIC since I think this is mostly all about HTTP layer components)